### PR TITLE
Ability to pay with credit card for BankTransfer bookings

### PR DESCRIPTION
**Pull request by Raibo**
.

### CODE REVIEW

File: Api/Services/Payments/IPaymentsService.cs [hunk 1]
```diff
@@ -12,5 +12,7 @@ public interface IPaymentsService
         Task<Result> ProcessPaymentChanges(Payment payment);
         
         Task<Result<(int AgentId, int AgencyId)>> GetServiceBuyer(string referenceCode);
+
+        Task<Result> SetPaymentMethodToCreditCard(string referenceCode);
```
Comment (mingazhev) [inline Api/Services/Payments/IPaymentsService.cs:16]: Should this be a separate operation?
I thought that this should be processed during credit card payment

Comment (Raibo) [reply Api/Services/Payments/IPaymentsService.cs:16]: This is a separate operation, which is called during credit card operation

File: Api/Services/Accommodations/Bookings/BookingPaymentService.cs [hunk 2]
```diff
@@ -248,6 +248,27 @@ public async Task<Result<(int AgentId, int AgencyId)>> GetServiceBuyer(string re
         }
 
 
+        public async Task<Result> SetPaymentMethodToCreditCard(string referenceCode)
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BookingPaymentService.cs:251]: I think this method will be more suitable for BookingRecordsManager and should look like "SetPaymentMethod(PaymentMethods paymentMethod)".
It just updates a booking field, nothing about payment processing

**Review by kirillta — APPROVED** [review]
Approved with @mingazhev comments in mind

**Review by mingazhev — APPROVED** [review]

**Pull request merged**